Apply-AssemblyVersionAndDefaults.ps1

```diff
@@ -0,0 +1,70 @@
+param (
+    [Parameter(Mandatory = $true)] [string] $buildNumber,
+    [Parameter(Mandatory = $true)] [string] $solutionDirectory,
+    [Parameter(Mandatory = $true)] [string] $UKHOAssemblyCompany,
+    [Parameter(Mandatory = $true)] [string] $UKHOAssemblyCopyright,
+    [Parameter(Mandatory = $true)] [string] $UKHOAssemblyVersionPrefix,
+    [Parameter(Mandatory = $true)] [string] $UKHOAssemblyProduct
+)
+
+$buildNumberRegex = "(.+)_([0-9]{3,5}).([0-9]{1,2})"
+$validBuildNumber = $buildNumber -match $buildNumberRegex
+
+if ($validBuildNumber -eq $false) {
+    Write-Error "Build number passed in must be in the following format: (BuildDefinitionName)_(ProjectVersion).(date:yyyyMMdd)(rev:.r)"
+    return
+}
+
+# Magic var $Matches comes from the above regex match statement: $buildNumber -match $buildNumberRegex
+$buildRevisionNumber = $Matches.2 + "." + $Matches.3
+$versionToApply = $UKHOAssemblyVersionPrefix + $buildRevisionNumber
+
+$UKHOAssemblyCopyright = "$UKHOAssemblyCopyright $(Get-Date -Format "yyyy")"
+
+$assemblyValues = @{
+    "Company"         = $UKHOAssemblyCompany;
+    "Copyright"       = $UKHOAssemblyCopyright;
+    "Description"     = $UKHOAssemblyProduct;
+    "Product"         = $UKHOAssemblyProduct;
+    "AssemblyVersion" = $versionToApply;
+    "FileVersion"     = $versionToApply;
+    "Version"         = $versionToApply;
```

Comment:
I was looking into CI build for value of `versionToApply`. 
For example version number generated is `1.0.20210.21` for build https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_build/results?buildId=7537&view=logs&j=4e6201ad-7029-5abf-ddbd-3fcb13fcbeca&t=474823bb-dd3a-55ba-99c4-d7022bf41484 

IMHO version number should be easy to link to it's CI build. I found it difficult to link it back, please can you have a look.

Reply:
Changes done now version is coming "1.0.20210524.6" where 1.0. is prefix version and 20210524.6 is date and rev for pipeline

Reply:
Assembly version has limitation that any major, minor, build, revision can store only till 65,535 so yyyymmdd is crossing the limit. I have change build number to $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yy)$(DayOfYear).$(BuildCounter)
now it comes like UKHO.exchange-set-service_merge_21144.8 and same value is added in assembly info